httpx/_config.py

```diff
@@ -218,10 +218,10 @@ def __init__(
             assert read is UNSET
             assert write is UNSET
             assert pool is UNSET
-            self.connect = timeout.connect  # type: typing.Optional[float]
-            self.read = timeout.read  # type: typing.Optional[float]
-            self.write = timeout.write  # type: typing.Optional[float]
-            self.pool = timeout.pool  # type: typing.Optional[float]
+            self.connect: typing.Optional[float] = timeout.connect
+            self.read: typing.Optional[float] = timeout.read
+            self.write: typing.Optional[float] = timeout.write
+            self.pool: typing.Optional[float] = timeout.pool
```

Comment:
Regardless of any linting enforcement or not, these typing changes are *clearly* a neater style. I'd be very much in favour of an isolated PR that just deals with cleaning up any "type as comment" cases that we're currently doing.